File: cpp/5-Longest_Palindromic_Substring.cpp
class Solution {
public:
    int findlength(string s, int left, int right){
        while(left>=0 && right<s.size() && s[left]==s[right]){
            left--;
            right++;
        }
        return right-left-1;
    }
    
    string longestPalindrome(string s) {
        if(s.size()==0) return "";
        int start=0, end=0;
        for(int i=0;i<s.size();i++){
            int l1=findlength(s,i,i);
            int l2=findlength(s,i,i+1);
            int l=max(l1,l2);        
            if(l>end-start+1){
                start=i-(l-1)/2;
                end=i+l/2;
            }
        }
        return s.substr(start, end-start+1);
    }
};

File: py/5. Longest Palindromic Substring.py

class Solution(object):
    def longestPalindrome(self, s):
        n=len(s)
        tmp=[[0 for i in range(n)]for j in range(n)]
        i=0
        maxlen=1
        while i<n:
            tmp[i][i]=1
            i=i+1
        i=0
        start=0
        while i<n-1:
            if s[i]==s[i+1]:
                tmp[i][i+1]=1
                start=i
                maxlen=2
            i=i+1
        k=3
        while k<=n:
            i=0
            while i<(n-k+1):
                j=i+k-1
                if tmp[i+1][j-1]==1 and s[i]==s[j]:
                    tmp[i][j]=1
                    if k>maxlen:
                        start=i
                        maxlen=k
                i=i+1
            k=k+1
        
        return(s[start:start+maxlen])
                
            
            
        
        
